Cron/Providers/PayByLinkExpiredPaymentOrdersProvider.php

```diff
@@ -125,15 +125,16 @@ protected function getExpiredOrders($expiredOrderIds): array
             ->setValue(Order::STATE_NEW)
             ->create();
 
-        $orderIdFilter = $this->filterBuilder->setField('state')
+        $orderIdFilter = $this->filterBuilder->setField('entity_id')
             ->setConditionType('in')
             ->setValue($expiredOrderIds)
             ->create();
 
-        $filterGroup = $this->filterGroupBuilder->setFilters([$stateFilter, $orderIdFilter])->create();
```

Comment:
Why this line were causing the issue?

`In Magento 2 all the filters in same FilterGroup will be added using OR operator. But all the FilterGroup will be added using AND operator.`